plugins/modules/proxysql_mysql_users.py

```diff
@@ -34,8 +35,14 @@
     description:
       - Encryption method used when I(encrypt_password) is set to C(True).
     type: str
-    choices: [ "mysql_native_password" ]
-    default: mysql_native_password
+    choices: [ "mysql_native_password", "caching_sha2_password" ]
+    default: caching_sha2_password
+  salt:
+    description:
+      - Salt used when encryption_method is set to C(caching_sha2_password).
+        If omitted the I(salt) is a randomly generated string and will change
+        value in proxysql db on each execution.
+    type: str
```

Comment:
```suggestion
    choices: [ "mysql_native_password", "caching_sha2_password" ]
    default: mysql_native_password
  salt:
    description:
      - Salt used when encryption_method is set to C(caching_sha2_password).
        If omitted the I(salt) is a randomly generated string and will change
        value in proxysql db on each execution.
    type: str
    version_added: 1.7.0
```

I don't think we should change the default value withing the proxysql collection release 1.X.X.  
But we can announce the switch for 2.0.0

Reply:
Yes you are right we can go for this